content/master/concepts/compositions.md

```diff
@@ -0,0 +1,993 @@
+  generated by the managed resources.
+* Using [Composite functions](#composite-functions) to allow custom programs to
+  run alongside the Composition. 
+* Creating a 
+  [custom definition of when a resource is ready](#resource-readiness-checks) to use. 
+ 
+###  Resource templates
+The 
+{{<hover label="resources" line="4">}}resources{{</hover>}} field of a 
+Composition's 
+{{<hover label="resources" line="3">}}spec{{</hover>}} 
+defines the set of things that a Composite Resource can create.
+
+{{<hint "note" >}}
+Read more about Composite Resources in the 
+[Composite Resources page]({{<ref "./composite-resources" >}}). 
+{{< /hint >}}
+  
+
+For example, a Composition can define a template to create a VM and an
+associated storage bucket at the same time. 
+
+The {{<hover label="resources" line="4">}}resources{{</hover>}} field lists the 
+individual resources with a 
+{{<hover label="resources" line="5">}}name{{</hover>}}.  
+This name identifies the
+resource inside the Composition and isn't related to the external name used with
+the Provider.
+
+#### Template a managed resource
+The contents of the 
+{{<hover label="resources" line="6">}}base{{</hover>}} are identical to
+creating a standalone [managed resource]({{<ref "./managed-resources">}}).
+
+This example uses 
+[Upbound's Provider AWS](https://marketplace.upbound.io/providers/upbound/provider-aws/v0.35.0)
+to define a 
+S3 storage {{<hover label="resources" line="8">}}Bucket{{</hover>}} and 
+EC2 compute {{<hover label="resources" line="15">}}Instance{{</hover>}}.
+
+After defining the 
+{{<hover label="resources" line="7">}}apiVersion{{</hover>}} and 
+{{<hover label="resources" line="8">}}kind{{</hover>}}, define the 
+{{<hover label="resources" line="10">}}spec.forProvider{{</hover>}} fields
+defining the settings of the resource.
+
+```yaml {copy-lines="none",label="resources"}
+apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+spec:
+  resources:
+    - name: StorageBucket
+      base:
+        apiVersion: s3.aws.upbound.io/v1beta1
+        kind: Bucket
+        spec:
+          forProvider:
+            region: "us-east-2"
+    - name: VM
+      base:
+        apiVersion: ec2.aws.upbound.io/v1beta1
+        kind: Instance
+        spec:
+          forProvider:
+            ami: ami-0d9858aa3c6322f73
+            instanceType: t2.micro
+            region: "us-east-2"
+```
+
+When a [Composite Resource]({{<ref "./composite-resources" >}}) uses this
+Composition template, the Composite Resource creates two new managed resources
+with all the provided 
+{{<hover label="resources" line="17">}}spec.forProvider{{</hover>}} settings.
+
+The {{<hover label="resources" line="17">}}spec.forProvider{{</hover>}}
+supports any settings used in a managed resource including
+applying `annotations` and `labels` or using a specific 
+`providerConfigRef`.
+
+{{<hint "note" >}}
+Compositions allow applying a `metadata.name` but ignores it. The
```

Comment:
Are you sure this is true? I thought they'd let you specify it.

Either way, I'd consider specifying your own `metadata.name` to be an anti-pattern. If it is actually possible you'll need to make sure they're somehow unique, so that three XRs using the same Composition don't try to make three composed MRs with the same name.

Reply:
In testing and what I could see from code that it's ignored.